### PR TITLE
1188 - block unblock endpoints

### PR DESCRIPTION
**Pull request by codyseibert**
Depends on #3285
POST@/cases/{caseId}/block
DELETE@/cases/{caseId}/block

### CODE REVIEW

File: shared/src/business/useCases/unblockCaseInteractor.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  BLOCK_CASE,
+  isAuthorized,
+} = require('../../authorization/authorizationClientService');
+
+/**
+ * used for unblocking a case
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {string} providers.caseId the caseId to block
```
Comment (rachaelparris) [inline shared/src/business/useCases/unblockCaseInteractor.js]: `caseId to unblock`

File: shared/src/business/useCases/unblockCaseInteractor.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const { unblockCaseInteractor } = require('./unblockCaseInteractor');
+const { MOCK_CASE } = require('../../test/mockCase');
+
+describe('unblockCaseInteractor', () => {
+  let applicationContext;
+
+  it('should update the case with the blocked flag set as true and attach a reason', async () => {
```
Comment (rachaelparris) [inline shared/src/business/useCases/unblockCaseInteractor.test.js]: Description needs to be udpated

File: web-api/src/cases/unblockCaseLambda.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const createApplicationContext = require('../applicationContext');
+const { getUserFromAuthHeader } = require('../middleware/apiGatewayHelper');
+const { handle } = require('../middleware/apiGatewayHelper');
+
+/**
+ * used for blocking a case
```
Comment (rachaelparris) [inline web-api/src/cases/unblockCaseLambda.js]: `used for unblocking a case`

**Review by rachaelparris — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

**Review by ericsorenson — APPROVED** [review]

**Pull request merged**